py3status/modules/volume_status.py

```diff
@@ -135,20 +130,107 @@ def _get_muted(self, output):
         else:
             return False
 
-    # this method is ran by py3status
+    # this method is called by py3status
     # returns a response dict
-    def current_volume(self):
+    def get_volume(self):
 
         # call amixer
-        output = check_output(shlex.split('amixer -D {} sget {}'.format(
-            self.device, self.channel))).decode('utf-8')
+        output = check_output(['amixer', '-D', self.device, 'sget', self.channel]).decode('utf-8')
 
         # get the current percentage value
         perc = self._get_percentage(output)
 
         # get info about channel mute status
         muted = self._get_muted(output)
 
+        return perc, muted
+
+    def volume_up(self, delta):
+        # volume up
+        self.run_cmd(self.cmd + ['{}%+'.format(delta)])
+
+    def volume_down(self, delta):
+        # volume down
+        self.run_cmd(self.cmd + ['{}%-'.format(delta)])
+
+    def toggle_mute(self):
+        # toggle mute
+        self.run_cmd(self.cmd + ['toggle'])
+
+
+class PulseaudioBackend(AudioBackend):
+    def __init__(self, device, channel):
+        AudioBackend.__init__(self, device, channel)
+        if self.device == 'default':
+            self.device = "0"
+        self.cmd = ["pamixer", "--sink", self.device]
+
+    def get_volume(self):
+        perc = check_output(self.cmd + ["--get-volume"]).decode('utf-8').strip()
+        muted = (self.run_cmd(self.cmd + ["--get-mute"]) == 0)
+        return perc, muted
+
+    def volume_up(self, delta):
+        # volume up
+        self.run_cmd(self.cmd + ["-i", str(delta)])
+
+    def volume_down(self, delta):
+        # volume down
+        self.run_cmd(self.cmd + ["-d", str(delta)])
+
+    def toggle_mute(self):
+        # toggle mute
+        self.run_cmd(self.cmd + ["-t"])
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    backend = 'alsa'
+    button_down = 0
+    button_mute = 0
+    button_up = 0
+    cache_timeout = 10
+    channel = 'Master'
+    device = 'default'
+    format = u'♪: {percentage}%'
+    format_muted = u'♪: muted'
+    threshold_bad = 20
+    threshold_degraded = 50
+    volume_delta = 5
+
+    def post_config_hook(self):
+        if self.backend == 'alsa':
+            self.backend = AlsaBackend(self.device, self.channel)
```

Comment:
Why not just `self.backend = AlsaBackend(self)`?

Personally I think you should use a different name not overwrite `backend`